mvc/src/main/java/nextstep/mvc/view/JsonView.java

```diff
@@ -1,12 +1,28 @@
 package nextstep.mvc.view;
 
+import com.fasterxml.jackson.databind.ObjectMapper;
 import jakarta.servlet.http.HttpServletRequest;
 import jakarta.servlet.http.HttpServletResponse;
 import java.util.Map;
+import java.util.Optional;
+import nextstep.web.support.MediaType;
 
 public class JsonView implements View {
 
+    private static final ObjectMapper objectMapper = new ObjectMapper();
+
     @Override
     public void render(Map<String, ?> model, HttpServletRequest request, HttpServletResponse response) throws Exception {
+        response.setContentType(MediaType.APPLICATION_JSON_UTF8_VALUE);
+
+        if (model.size() <= 1) {
+            Optional<?> firstValue = model.values().stream()
+                .findFirst();
+            String jsonValue = objectMapper.writeValueAsString(firstValue.get());
+            response.getWriter().write(jsonValue);
+            return;
+        }
+        String value = objectMapper.writeValueAsString(model);
+        response.getWriter().write(value);
```

Comment:
```suggestion
        PrintWriter writer = response.getWriter();

        if (model.size() <= 1) {
            Object object = model.values().stream()
                    .findFirst()
                    .orElseThrow(() -> new IllegalArgumentException("렌더링할 값이 없습니다."));
            String jsonValue = objectMapper.writeValueAsString(object);
            writer.write(jsonValue);
            return;
        }
        String value = objectMapper.writeValueAsString(model);
        writer.write(value);
```
0개 일 때 예외를 발생시키도록 하거나(위의 suggestion), 아니면 `Writer`가 아무 것도 쓰지 않고 return하도록 바꾸는 것은 어떨까요? 조건문은 1개 이하(model.size() <=1)일 때이지만, 코드는 마치 1개일 때(model.size() == 1) 작동하는 것처럼 작성되어 있어 건의드려봅니다.